src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -285,10 +288,62 @@ partial class BoundPropertyAccess : IPropertyReferenceExpression
         IPropertySymbol IPropertyReferenceExpression.Property => this.PropertySymbol;
        
         IExpression IMemberReferenceExpression.Instance => this.ReceiverOpt;
-       
+
+        ISymbol IMemberReferenceExpression.Member => this.PropertySymbol;
+
         protected override OperationKind ExpressionKind => OperationKind.PropertyReferenceExpression;
     }
 
+    partial class BoundEventAccess : IEventReferenceExpression
+    {
+        IEventSymbol IEventReferenceExpression.Event => this.EventSymbol;
+
+        IExpression IMemberReferenceExpression.Instance => this.ReceiverOpt;
+
+        ISymbol IMemberReferenceExpression.Member => this.EventSymbol;
+
+        protected override OperationKind ExpressionKind => OperationKind.EventReferenceExpression;
+    }
+
+    partial class BoundEventAssignmentOperator : IEventAssignmentExpression
+    {
+
+        IEventSymbol IEventAssignmentExpression.Event => this.Event;
+
+        IExpression IEventAssignmentExpression.EventInstance => this.ReceiverOpt;
+
+        IExpression IEventAssignmentExpression.HandlerValue => this.Argument;
+
+        bool IEventAssignmentExpression.Adds => this.IsAddition;
+
+        protected override OperationKind ExpressionKind => OperationKind.EventAssignmentExpression;
+    }
+
+    partial class BoundDelegateCreationExpression : IMethodBindingExpression
+    {
+        IExpression IMemberReferenceExpression.Instance
```

Comment:
If you wanted to abuse new language features this could be reduced (I hesitate to say 'simplified' to):

``` C#
IExpression IMemberReferenceExpression.Instance => (this.Argument as BoundMethodGroup)?.InstanceOpt;
```